platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -375,28 +389,37 @@ private String getEntityId() {
     return String.format("https://%s:%s%s/saml", hostname, port, rootContext);
   }
 
-  private void logout() {
-    HttpSession session = sessionFactory.getOrCreateSession(request);
+  private SecurityAssertion getIdpSecurityAssertion() {
 
-    SecurityTokenHolder tokenHolder =
-        ((SecurityTokenHolder) session.getAttribute(SecurityConstants.SAML_ASSERTION));
+    return new SecurityAssertionImpl(getTokenHolder().getSecurityToken(IDP_REALM_NAME));
```

Comment:
❓ can `getTokenHolder` or `getSecurityToken` return null? If so this could throw a NPE

Reply:
I don't know. There were no `null` checks previously, so I didn't add any in. I just used Intellij's "extract method" to create helper methods.

Reply:
Checked the code and both `getTokenHolder()` can return null (so we should protect against that) and `getSecurityToken(..)` can return null as well ( which would throw an NPE when passed to `new SecurityAssertionImpl(..)`

Reply:
I still believe we need to ensure the token returned from `getSecurityToken(IDP_REALM_NAME)` is not null before passing it to the `SecurityAssertionImpl` constructor. Perhaps in this case we just would want to throw a more descriptive error than the NPE that would be thrown otherwise.

Also since getTokenHolder is nullable we need to verify that is not null here either.